### ISSUE TITLE
VAE

### ISSUE THREAD

**Issue by MasterSPP** [UTokyo-FieldPhenomics-Lab/DODA#5]
hello!, I am amazed by your work. I now want to train my own VAE model, but I don’t have the training code. Can you provide the code for training VAE?

**Comment by illrayy** [comment]
You can simply modify the config path in train_wheat.py to `configs/autoencoder/DODA_wheat_autoencoder_kl_64x64x3.yaml
`

OR you can use [this](https://github.com/illrayy/latent-diffusion/tree/main) repository to train the VAE and LDM. This repository's environment is compatible with DODA, but I haven't tested this repository, so if you encounter any issues feel free to report them.

Training of VAE can be started by running
`CUDA_VISIBLE_DEVICES=<GPU_ID> python main.py --base configs/autoencoder/autoencoder_kl_64x64x3_wheat.yaml  --scale_lr False -t --gpus 0,  `
Training of LDM can be started by running
`CUDA_VISIBLE_DEVICES=<GPU_ID> python main.py --base configs/autoencoder/DODA_wheat_ldm_kl_4.yaml  --scale_lr False -t --gpus 0, `

**Comment by MasterSPP** [comment]
tanks！

**Comment by MasterSPP** [comment]
When training VAE, I did not make any changes to the configuration file, logger_freq = 5000
max_steps = 80000, sd_locked = True
learning_rate = 2.5e-6
accumulate_grad_batches = 2 But in the first two epochs of training, aeloss_step=6.53e+4, discloss_step=0.000, is this normal? How long does it take to train?

**Comment by illrayy** [comment]
It's normal, the `disc_start: 50001` in config file controls when disc loss is added to the training, before that the disc-loss is 0.

I trained VAE for 170k iterations(global steps), and cost 7 V100 days
